src/mpid/ch4/src/ch4_progress.h

```diff
@@ -317,34 +259,9 @@ MPL_STATIC_INLINE_PREFIX int MPID_Stream_progress(MPIR_Stream * stream_ptr)
     return mpi_errno;
 }
 
-#if MPICH_THREAD_GRANULARITY == MPICH_THREAD_GRANULARITY__GLOBAL
-#define MPIDI_PROGRESS_YIELD() MPID_THREAD_CS_YIELD(GLOBAL, MPIR_THREAD_GLOBAL_ALLFUNC_MUTEX)
-#else
-#define MPIDI_PROGRESS_YIELD() MPID_Thread_yield()
-#endif
-
 MPL_STATIC_INLINE_PREFIX int MPID_Progress_wait(MPID_Progress_state * state)
 {
-    int mpi_errno = MPI_SUCCESS;
-
-    MPIR_FUNC_ENTER;
-
-    state->progress_made = 0;
-    while (1) {
-        mpi_errno = MPIDI_progress_test(state, 1);
-        MPIR_ERR_CHECK(mpi_errno);
-        if (state->progress_made) {
-            break;
-        }
-        MPIDI_PROGRESS_YIELD();
-    }
-
-    MPIR_FUNC_EXIT;
-
-  fn_exit:
-    return mpi_errno;
-  fn_fail:
-    goto fn_exit;
+    return MPID_Progress_test(state);
```

Comment:
Now that progress_wait is really just progress_test, do we need the first 2 commits that for MPID_Progress_test with a macro #define?

Reply:
Not really :)

Reply:
OK, I'll let you decide if you want to remove them before merge. I'm approving either way.

Reply:
> OK, I'll let you decide if you want to remove them before merge. I'm approving either way.

I'll remove them

Reply:
This is 2 nodes of skylake with ch4:ofi. FWIW, I find this test to be pretty inconsistent. Run to run I get results that differ 5% or more.
old:
```
Thread 0 message rate below threshold: 2.54 / 2.99 = 0.85 (threshold = 0.05)
Thread 1 message rate below threshold: 2.59 / 2.99 = 0.87 (threshold = 0.05)
Thread 2 message rate below threshold: 2.64 / 2.99 = 0.88 (threshold = 0.05)
Thread 3 message rate below threshold: 2.61 / 2.99 = 0.87 (threshold = 0.05)
```
new:
```
Thread 0 message rate below threshold: 2.38 / 3.01 = 0.79 (threshold = 0.05)
Thread 1 message rate below threshold: 2.36 / 3.01 = 0.78 (threshold = 0.05)
Thread 2 message rate below threshold: 2.34 / 3.01 = 0.78 (threshold = 0.05)
Thread 3 message rate below threshold: 2.37 / 3.01 = 0.79 (threshold = 0.05)
```